addons/website_sale_autocomplete/static/src/js/addressForm.js

```diff
@@ -0,0 +1,88 @@
+/** @odoo-module */
+
+import publicWidget from 'web.public.widget';
+import { DropPrevious } from 'web.concurrency';
+import { debounce } from "@web/core/utils/timing";
+
+
+publicWidget.registry.AddressForm = publicWidget.Widget.extend({
+    selector: '.oe_cart .checkout_autoformat:has(input[name="street"][data-autocomplete-enabled="1"])',
+    events: {
+        'input input[name="street"]': '_onChangeStreet',
+        'click .js_autocomplete_result': '_onClickAutocompleteResult'
+    },
+
+    init: function() {
+        this.streetAndNumberInput = document.querySelector('input[name="street"]');
+        this.cityInput = document.querySelector('input[name="city"]');
+        this.zipInput = document.querySelector('input[name="zip"]');
+        this.countrySelect = document.querySelector('select[name="country_id"]');
+        this.stateSelect = document.querySelector('select[name="state_id"]');
+        this.dp = new DropPrevious();
+
+        this._onChangeStreet = debounce(this._onChangeStreet, 200);
+        this._super.apply(this, arguments);
+    },
+
+    _onChangeStreet: function(ev) {
+        this.dp.add(
+            this._rpc({
+                route: '/autocomplete/address',
+                params: {
+                    partial_address: ev.currentTarget.value
+                }
+            })).then((response) => {
+            if(response && response.template && response.template.trim()){
+                const inputContainer = ev.currentTarget.parentNode;
+                const dropdown = inputContainer.querySelector('.dropdown-menu');
+                if(dropdown) dropdown.remove();
+                inputContainer.appendChild($(response.template)[0]);
+            }
+        })
+    },
+
+    _onClickAutocompleteResult: function(ev) {
+        const dropDown = ev.currentTarget.parentNode;
+
+        this._rpc({
+            route: '/autocomplete/address_full',
+            params: {
+                address: ev.currentTarget.innerText,
+                google_place_id: ev.currentTarget.dataset.googlePlaceId
+            }
+        }).then((result) => {
+            if(result.address){ // Don't change this field in case something goes wrong
```

Comment:
Assigning repeatedly accessed sub-items is nice to avoid having to repeat them over and over.

Also guards to avoid excessive right drift.